moesifasgi/middleware.py

```diff
@@ -166,75 +154,93 @@ async def dispatch(self, request, call_next):
-                else:
-                    if self.DEBUG:
-                        logger.info('Skipped Event as the moesif event model is None')
+        # Check if needs to skip the event
+        skip = await self.logger_helper.should_skip(self.settings, request, response, self.DEBUG)
+        if skip:
+            if self.DEBUG:
+                logger.info("Skipped Event using should_skip configuration option")
+            return response
+
+        # Read Response Body
+        resp_body = None
+        if self.LOG_BODY:
+            # Consuming FastAPI response and grabbing body here
+            resp_body = [section async for section in response.__dict__['body_iterator']]
+            # Preparing FastAPI response
+            response.__setattr__('body_iterator', async_iterator_wrapper(resp_body))
+
+        # Prepare Event Response Model
+        event_rsp = self.event_mapper.to_response(response, response_time, resp_body)
+
+        # Add user, company, session_token, and metadata
+        user_id = await self.logger_helper.get_user_id(self.settings, request, response, dict(request.headers), self.DEBUG)
+        company_id = await self.logger_helper.get_company_id(self.settings, request, response, self.DEBUG)
+        session_token = await self.logger_helper.get_session_token(self.settings, request, response, self.DEBUG)
+        metadata = await self.logger_helper.get_metadata(self.settings, request, response, self.DEBUG)
+
+        # Prepare Event Model
+        event_data = await self.event_mapper.to_event(event_req, event_rsp, user_id, company_id, session_token, metadata, blocked_by)
+
+        # Mask Event Model
+        if self.logger_helper.is_coroutine_function(self.logger_helper.mask_event):
+            event_data = await self.logger_helper.mask_event(event_data, self.settings, self.DEBUG)
+        else:
+            event_data = self.logger_helper.mask_event(event_data, self.settings, self.DEBUG)
+
+        # Sampling percentage
+        random_percentage = random.random() * 100
+        self.sampling_percentage = self.config.get_sampling_percentage(event_data, user_id, company_id)
+
+        # Add Weight to the event
+        event_data.weight = 1 if self.sampling_percentage == 0 else math.floor(100 / self.sampling_percentage)
+
+        if random_percentage >= self.sampling_percentage:
+            logger.debug(f"Skipped Event due to sampling percentage: {str(self.sampling_percentage)}"
+                         f" and random percentage: {str(random_percentage)}")
+            return response
+
+        try:
+            # Add Event to the queue if able and count the dropped event if at capacity
+            if self.worker_pool.add_event(event_data):
+                logger.debug("Add Event to the queue")
+                if self.DEBUG:
+                    logger.debug(f"Event added to the queue: {APIHelper.json_serialize(event_data)}")
```

Comment:
any log protected by `self.debug` should be `info` level otherwise it defeats the purpose of moesif/protected/debug logging.